sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -933,7 +935,7 @@ class Analyzer(
               v1SessionCatalog.getRelation(v1Table.v1Table)
             case table =>
               SubqueryAlias(
-                identifier,
+                ident.asMultipartIdentifier,
```

Comment:
shall we add catalog name too? to support cases like `select spark_catalog.default.t.i from t`.

Reply:
We could, but it will fallback to the new column resolution algo: https://github.com/apache/spark/pull/27391#discussion_r383091921. I was thinking about adding this support when I update the resolution rule.

Reply:
ah ok. Let's leave it for now.